pkg/runner/runner.go

```diff
@@ -221,21 +248,39 @@ func (runner *Runner) initializeMainEntrypoint() (mem.MemoryAddress, error) {
 		return mem.UnknownAddress, err
 	}
 	switch runner.runnerMode {
-	case ExecutionModeZero, ExecutionModeCairo, ProofModeCairo:
+	case ExecutionModeZero, ExecutionModeCairo:
 		returnFp := memory.AllocateEmptySegment()
 		mvReturnFp := mem.MemoryValueFromMemoryAddress(&returnFp)
-		if runner.runnerMode == ProofModeCairo {
-			// In Cairo mainPCOffset is equal to the offset of program segment base
-			return runner.initializeEntrypoint(uint64(0), nil, &mvReturnFp, memory, stack, 2)
-		} else if runner.runnerMode == ExecutionModeCairo {
-			return runner.initializeEntrypoint(uint64(0), nil, &mvReturnFp, memory, stack, 0)
+		if runner.runnerMode == ExecutionModeCairo {
+			return runner.initializeEntrypoint(uint64(0), nil, &mvReturnFp, memory, stack)
 		} else {
 			mainPCOffset, ok := runner.program.Entrypoints["main"]
 			if !ok {
 				return mem.UnknownAddress, errors.New("can't find an entrypoint for main")
 			}
-			return runner.initializeEntrypoint(mainPCOffset, nil, &mvReturnFp, memory, stack, 0)
+			return runner.initializeEntrypoint(mainPCOffset, nil, &mvReturnFp, memory, stack)
+		}
```

Comment:
Perhaps it's better to separate these two, at the end you have the logic in different branches already. I think it will be more readable this way. Should we re-add the comment regarding `mainPCOffset` in Cairo?

Reply:
did a little cleanup, moved all the runner modes into separate switch branches. Also removed the assignment of "__start__" label in AssembleProgram (which is always 0 for cairo), and hardcoded the 0 value directly in the initialization.